iterables/src/main/java/com/palantir/logsafe/Iterables.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright (C) 2007 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.palantir.logsafe;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Iterators;
+import java.util.Collection;
+
+public final class Iterables {
+    public static final String DEFAULT_ERROR_MESSAGE = "Expected exactly one element in collection";
+
+    private Iterables() {}
+
+    public static <T, C extends Collection<T>> T getOnlyElement(C collection, Arg<?>... args) {
```

Comment:
That's a collection, not an iterable. Guava Iterables takes an iterable, and iterators takes an iterator.

Reply:
Yeah looks like i mixed the two together -- I'd be happy to implement this for both Iterables and Iterators